IdentityCore/src/validation/MSIDAuthority.m

```diff
@@ -239,6 +239,17 @@ - (void)loadOpenIdMetadataWithContext:(nullable id<MSIDRequestContext>)context
      }];
 }
 
+- (BOOL)isSameEnvironmentAsAuthority:(MSIDAuthority *)authority
+{
+    NSArray<NSString *> *aliases = [self defaultCacheEnvironmentAliases];
+    for (NSString *environment in aliases)
+    {
+        if ([authority.environment isEqualToString:environment]) return YES;
```

Comment:
This check will fail is authority is nil. let's add in a check before line 244 to make sure authority object is not nil and also it has a valid environment object

Reply:
if the authority is nil, the whole condition of line 247 will be false, no? :)

Reply:
yeah but wouldn't it cause runtime error when you try to query property from a nil object? :)

Reply:
Objective C is special, haha. See here: https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ObjectiveC/Chapters/ocObjectsClasses.html (search for "send a message to nil") :)

Reply:
True, but I had an issue in the past when the object was NSNull object. then it would throw an error. I think it is still a good idea to add a check against nil, but it's up to you.

Reply:
NSNull is a good point. Added sanity check as well as updating UT :)